src/main/java/org/kuali/kfs/gl/batch/service/impl/PosterServiceImpl.java

```diff
@@ -0,0 +1,1308 @@
+                    int automatedEntriesCount = automatedEntries.size();
+
+                    if (automatedEntriesCount > 0) {
+                        for (Iterator icrIter = automatedEntries.iterator(); icrIter.hasNext();) {
+                            IndirectCostRecoveryRateDetail icrEntry = (IndirectCostRecoveryRateDetail) icrIter.next();
+                            KualiDecimal generatedTransactionAmount = null;
+
+                            if (!icrIter.hasNext()) {
+                                generatedTransactionAmount = distributionAmount;
+
+                                // Log differences that are over WARNING_MAX_DIFFERENCE
+                                if (getPercentage(transactionAmount, icrEntry.getAwardIndrCostRcvyRatePct()).subtract(distributionAmount).abs().isGreaterThan(WARNING_MAX_DIFFERENCE)) {
+                                    List<Message> warnings = new ArrayList<Message>();
+                                    warnings.add(new Message("ADJUSTMENT GREATER THAN " + WARNING_MAX_DIFFERENCE, Message.TYPE_WARNING));
+                                    reportWriterService.writeError(et, warnings);
+                                }
+                            }
+                            else if (icrEntry.getTransactionDebitIndicator().equals(KFSConstants.GL_DEBIT_CODE)) {
+                                generatedTransactionAmount = getPercentage(transactionAmount, icrEntry.getAwardIndrCostRcvyRatePct());
+                                distributionAmount = distributionAmount.add(generatedTransactionAmount);
+                            }
+                            else if (icrEntry.getTransactionDebitIndicator().equals(KFSConstants.GL_CREDIT_CODE)) {
+                                generatedTransactionAmount = getPercentage(transactionAmount, icrEntry.getAwardIndrCostRcvyRatePct());
+                                distributionAmount = distributionAmount.subtract(generatedTransactionAmount);
+                            }
+                            else {
+                                // Log if D / C code not found
+                                List<Message> warnings = new ArrayList<Message>();
+                                warnings.add(new Message("DEBIT OR CREDIT CODE NOT FOUND", Message.TYPE_FATAL));
+                                reportWriterService.writeError(et, warnings);
+                            }
+                            //KFSMI-5614 CHANGED
+                            generateTransactionsBySymbol(et, icrEntry, generatedTransactionAmount, runDate, OUTPUT_GLE_FILE_ps, icrGenerationMetadata);
+
+                            reportOriginEntryGenerated = reportOriginEntryGenerated + 2;
+                        }
+                    }
+                    // Delete expenditure record
+                    expenditureTransactionDao.delete(et);
+                    reportExpendTranDeleted++;
+
+                }
+                catch (RuntimeException re) {
```

Comment:
I had to include the foundation's PosterServiceImpl class because of this line. In order to pass an Exception subclass up when no suitable ICR Account is found (which then writes adds that error info for the line to the log and continues as seen in the Exception catch), we need to subclass Exception rather than RuntimeException. In order to throw something that is not a RuntimeException within a method, you must either catch it and handle it or declare that your method throws it. However, if a superclass doesn't declare that it throws this exception, your subclass can't either. So I also needed to change retrieveSubAccountIndirectCostRecoveryMetadata and retrieveAccountIndirectCostRecoveryMetadata so that they declared the throw because each of those methods calls the CuIndirectCostRecoveryAccountDistributionMetadata constructor when inserting a new item in the list. That constructor must be used in order for all this to work, unfortunately (can't just use the stock IndirectCostRecoveryAccountDistributionMetadata).

Alternatively, I could have only changed retrieveSubAccountIndirectCostRecoveryMetadata and retrieveAccountIndirectCostRecoveryMetadata to add the throw declaration, and changed the constructor call in a pair of overridden methods in CuPosterImpl. I did not do this because it seems like it would make it more difficult to hunt down what actually changed in the class. As it is, you should be able to do a diff and see the difference fairly quickly.